Nil pointer on early shutdown on windows

**Describe the bug**
There is a nil pointer dereference when the visor is shut down after around 1 second after its startup. Works fine before and after this timeframe. This is only seen on windows so far.

**Environment information:**

- OS: Windows

**Steps to Reproduce**
Steps to reproduce the behavior:

1. Start visor `.\skywire-visor.exe -c .\skywire-config.json`
2. After about 1 sec close the visor with `Ctrl+C`
3. See error (try multiple times if no error)

**Actual behavior**
```
[2022-05-13T17:57:27+05:30] INFO [dmsg_ctrl]: Initialized in 1.1989295s (1.6592612s with dependencies)
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x60 pc=0xe8efc9]

goroutine 61 [running]:
github.com/james-barrow/golang-ipc.(*Server).Write(0x0, 0x0, {0x1b2b260, 0x0, 0x0})
        D:/Code/go/skywire/vendor/github.com/james-barrow/golang-ipc/server_all.go:242 +0x29
github.com/skycoin/skywire/pkg/app/appserver.(*Proc).Stop(0xc0002d2480)
        D:/Code/go/skywire/pkg/app/appserver/proc.go:263 +0x6f
github.com/skycoin/skywire/pkg/app/appserver.(*procManager).stopAll(0xc00028f300)
        D:/Code/go/skywire/pkg/app/appserver/proc_manager.go:349 +0xf2
github.com/skycoin/skywire/pkg/app/appserver.(*procManager).Close(0xc00028f300)
        D:/Code/go/skywire/pkg/app/appserver/proc_manager.go:374 +0xdc
github.com/skycoin/skywire/pkg/visor.(*Visor).Close.func1({{0x105c8d1, 0x1553e20}, 0xc000004ed0})
        D:/Code/go/skywire/pkg/visor/visor.go:222 +0x2f
created by github.com/skycoin/skywire/pkg/visor.(*Visor).Close
        D:/Code/go/skywire/pkg/visor/visor.go:221 +0x456
```

--- a/pkg/app/appserver/proc.go
+++ b/pkg/app/appserver/proc.go
@@ -31,10 +31,11 @@ var (
 // communication.
 // TODO(evanlinjin): In the future, we will implement the ability to run multiple instances (procs) of a single app.
 type Proc struct {
-	ipcServer *ipc.Server
-	disc      appdisc.Updater // app discovery client
-	conf      appcommon.ProcConfig
-	log       *logging.Logger
+	ipcServer   *ipc.Server
+	ipcServerWg sync.WaitGroup
+	disc        appdisc.Updater // app discovery client
+	conf        appcommon.ProcConfig
+	log         *logging.Logger
 
 	logDB appcommon.LogStore
 
@@ -93,6 +94,10 @@ func NewProc(mLog *logging.MasterLogger, conf appcommon.ProcConfig, disc appdisc
 		connCh:  make(chan struct{}, 1),
 		m:       m,
 		appName: appName,
+	}
+
+	if runtime.GOOS == "windows" {
+		p.ipcServerWg.Add(1)
 	}
 	return p
 }
@@ -225,9 +230,11 @@ func (p *Proc) Start() error {
 			if err != nil {
 				_ = p.cmd.Process.Kill() //nolint:errcheck
 				p.waitMx.Unlock()
+				p.ipcServerWg.Done()
 				return
 			}
 			p.ipcServer = ipcServer
+			p.ipcServerWg.Done()
 		}
 
 		// Wait for proc to exit.
@@ -260,8 +267,11 @@ func (p *Proc) Stop() error {
 				return err
 			}
 		} else {
-			if err := p.ipcServer.Write(skyenv.IPCShutdownMessageType, []byte("")); err != nil {
-				return err
+			p.ipcServerWg.Wait()
+			if p.ipcServer != nil {
+				if err := p.ipcServer.Write(skyenv.IPCShutdownMessageType, []byte("")); err != nil {
+					return err
+				}
 			}
 		}
 	}

